File: todo_api/models/todo.go
package models

type Todo struct {
	ID          string  `json:"id"`
	Title       string  `json:"title" binding:"required"`
	Description *string `json:"description,omitempty"`
	Completed   bool    `json:"completed"`
}

type CreateTodoRequest struct {
	Title       string  `json:"title" binding:"required"`
	Description *string `json:"description,omitempty"`
}

type UpdateTodoRequest struct {
	Title       *string `json:"title,omitempty"`
	Description *string `json:"description,omitempty"`
	Completed   *bool   `json:"completed,omitempty"`
}

File: todo-app/src/types/todo.ts
export interface Todo {
    id: string;
    name: string;
    description?: string;
    completed: boolean;
    deadline?: Date;
}

export interface CreateTodoRequest {
    title: string;
    description?: string;
    deadline?: Date;
}

export interface UpdateTodoRequest {
    title?: string;
    description?: string;
    completed?: boolean;
    deadline?: Date;
}